---
title: "2021_01_27_FleaMicrobiome"
author: "Charlotte Manvell"
date: "1/27/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r load packages, warning=FALSE}
library(dada2); packageVersion("dada2")
library(phyloseq); packageVersion("phyloseq")
library(here)
library(ggplot2)
library(filesstrings)
library(GEOquery)
library(ape)
library(vegan)
library(RVAideMemoire)
library(decontam); packageVersion("decontam")
```
Extract fastq files from tar file and set up path 

Set up to not run if script is run again as path is now set up
```{r, eval = FALSE}
untar(here("flea_main", "fastqfiles_MiSeq2_162_Breitschwerdt_1.tar"), exdir = here("flea_microbiome"))
untar(here("flea_main", "fastqfiles_MiSeq2_163_Breitschwerdt_2.tar"), exdir = here("flea_microbiome"))
filestomove <- c(list.files(here("flea_microbiome", "fastqfiles_MiSeq2_163_Breitschwerdt_2")))
filestomove2 <- c(list.files(here("flea_microbiome", "fastqfiles_MiSeq2_162_Breitschwerdt_1")))
file.move(here("flea_microbiome", "fastqfiles_MiSeq2_163_Breitschwerdt_2", filestomove), here("flea_microbiome"), overwrite = TRUE)
file.move(here("flea_microbiome", "fastqfiles_MiSeq2_162_Breitschwerdt_1", filestomove2), here("flea_microbiome"), overwrite = TRUE)
unlink(here("flea_microbiome", "fastqfiles_MiSeq2_163_Breitschwerdt_2"), recursive = TRUE)
unlink(here("flea_microbiome", "fastqfiles_MiSeq2_162_Breitschwerdt_1"), recursive = TRUE)
```
Set path 
```{r}
path <- here("flea_microbiome")
list.files(path)
```
Obtain matched lists of forward and reverse fastq files 
```{r}
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE)) #sort forward sequences
fnRs <- sort(list.files(path, pattern = "_R2_001.fastq", full.names = TRUE)) #sort reverse sequences
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1) #extract sample names
```

**Inspect read quality profiles**
Forward Reads
```{r}
plotQualityProfile(fnFs[1:2])
```
In order to create read quality plot the following may be run in an Rscript as it doesn't work in Rmarkdown for some unknown reason 
```{r, eval = FALSE}
lengthforward <- length(fnFs)
pdf(here("flea_main", "forwardread.pdf"))
for(i in 1:2){
  print(plotQualityProfile(fnFs[[i]]))
}
dev.off()
```


Reverse Reads
```{r}
plotQualityProfile(fnRs[1:2])
```
```{r, eval = FALSE}
lengthreverse <- length(fnRs)
pdf(here("flea_main", "reverseread.pdf"))
for(i in 1:lengthreverse){
  print(plotQualityProfile(fnRs[[i]]))
}
dev.off()
```
**Filtering**
Assign file names
```{r}
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```

```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,220),
              maxN=0, maxEE=c(5,5), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE, trimLeft = c(17,21)) # On Windows set multithread=FALSE
head(out)
```
**Error Rates**
Forward Error Rates
```{r}
errF <- learnErrors(filtFs, multithread = TRUE)
```
START HERE
Reverse Error Rates
```{r}
errR <- learnErrors(filtRs, multithread = TRUE)
```
Plot Forward Errors
```{r}
plotErrors(errF, nominalQ=TRUE)
```

**Sample Inference**

```{r}
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
```
```{r}
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
```
Merge sequences 
```{r}
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
head(mergers[[1]])
```

```{r}
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

```{r}
table(nchar(getSequences(seqtab)))
```
Remove chimeras 
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```
Percentage of sequences remaining when chimeras are removed
```{r}
(sum(seqtab.nochim)/sum(seqtab))*100
```
Track the number of reads in each phase of pipeline
```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
write.csv(track, here("flea_main", "tracked_sequences.csv"))
```
## Assign Taxonomy 
```{r}
taxa <- assignTaxonomy(seqtab.nochim, here("flea_main", "silva_nr_v132_train_set.fa.gz"), multithread = TRUE)
taxa.print <- taxa
rownames(taxa.print) <- NULL
head(taxa.print)
```

**Encode File Names**
```{r}
samples.out <- rownames(seqtab.nochim)
sample.names <- gsub("\\._*", "", samples.out)
flea.info <- read.csv(here("flea_main", "FleaInfo.csv"))
rownames(flea.info) <- flea.info[,1]
```

**Construct phyloseq object**
```{r}
ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE),
               sample_data(flea.info),
               tax_table(taxa))
dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
psProp <- ps %>%
  transform_sample_counts(function(otu) otu/sum(otu))
ps 
saveRDS(ps, here("flea_main", "Fleapsnew.rds")) ## MOVE TO asv_filt_decontam
```

## Filtering
### Decontam
**Library Size**
```{r, fig.width = 13, fig.height = 4}
df <- as.data.frame(sample_data(ps)) #create dataframe of sample data
df$LibrarySize <- sample_sums(ps)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
ggplot(df, aes(x = reorder(flea.id, LibrarySize), y = LibrarySize, color = Sample_or_Control))+
  geom_point()+
  theme(axis.text.x = element_text(angle=90, hjust = 1, size = 6), legend.position = "bottom")+
  labs(x = "Sample ID", color = "Sample or Control")+
  ggsave(filename = here("flea_main", "librarysize.pdf"), plot=last_plot(), width = 13, height = 4)
```
**Determine the number of contaminant ASVs**
```{r}
sample_data(ps)$is.neg <- sample_data(ps)$Sample_or_Control == "Control"
contamdf.prev <- isContaminant(ps, method = "prevalence", neg="is.neg")
table(contamdf.prev$contaminant) #True - contaminant, False - not contaminant 
```
```{r}
head(which(contamdf.prev$contaminant))
```

```{r}
ps.pa <- transform_sample_counts(ps, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "Control", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "True Sample", ps.pa)
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg), contaminant=contamdf.prev$contaminant)
ggplot(df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point()+
  labs(x = "Prevalence (Negative Controls)", y = "Prevalence (True Samples)", title = "Prevalence of ASVs in Samples vs. Negative Controls", color = "Contaminant Status")+
  ggsave(filename = here("flea_main", "ASV_prev_pos_neg.pdf"), plot=last_plot())
```

**Remove ASVs designated as contaminants by decontam package**
```{r}
ps.decon <- prune_taxa(!contamdf.prev$contaminant, ps)
ps.decon
```


**Remove negative control samples**
```{r}
ps.decon <- prune_samples(df$location != "Negative Control", ps.decon)
ps.deconProp <- ps.decon %>% 
  transform_sample_counts(function(otu) otu/sum(otu))
ps.decon
```


**OTUs not classified to phylum level**
```{r}
table(tax_table(ps.decon)[,"Phylum"], exclude = NULL)
```
**Remove OTUs from N/A phylum**
```{r}
ps.1 <- subset_taxa(ps.decon, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"), prune = TRUE)
ps.1Prop <- ps.1 %>% 
  transform_sample_counts(function(otu) otu/sum(otu))
ps.1
```

### Remove Common Contaminants

List of potential contaminants obtained from (Salter et al., 2014) and those known to occur in the flea microbiome were removed (Lawrence et al., 2015; Vasconcelos et al., 2018)

**Remove chloroplast**
```{r}
get_taxa_unique(ps.1, taxonomic.rank = "Order") %>%
  str_detect("Chloroplast") %>%
  table
ps.2 <- subset_taxa(ps.1, Order !="Chloroplast", prune=TRUE)
ps.2Prop <- ps.2 %>% 
  transform_sample_counts(function(otu) otu/sum(otu))
```

**Read contaminant csv**
```{r, fig.width = 20, fig.height = 8}
contamdf.salter <- read.csv(here("flea_main", "ContaminantsList.csv"))
removeContams <- contamdf.salter %>% #make list of contaminants that have not been reported in flea studies
  dplyr::filter(Genus!="unclassified",
         inFleaStudy==FALSE) %>%
  .$Genus %>%
  paste0(collapse="|")
ps.2Prop %>%
  subset_taxa(str_detect(Genus, removeContams))%>% # any potential contaminants
  psmelt() %>%
  ggplot(aes(x=flea.id, y=Abundance, fill=Phylum)) +
  geom_bar(stat="identity") +
  facet_grid(~location, scales="free_x", space = "free_x") +
  theme(axis.text.x = element_blank()) +
  scale_y_continuous(limits=c(0, 1)) +
  labs(title = "Proportion of potential contaminants")+
  ylim(0,0.2)
```

ps.3 <- subset_taxa(ps.2, !str_detect2(Genus, removeContams, default = FALSE), prune = TRUE) #need to update R to run cna't load package for str_detect2


```{r}
plot_richness(ps.2, measures=c("Shannon", "Simpson"), x="location", color="cat.sex")
```

```{r}
plot_richness(ps.2, measures=c("Shannon", "Simpson"), x="ssrA", color="location")+
  ggsave(here("diversityssrA.pdf"), plot = last_plot())
```

```{r}
ord.nmds.bray <- ordinate(ps.2Prop, method="NMDS", distance="bray")
plot_ordination(ps.2Prop, ord.nmds.bray, color="location", title="Bray NMDS")
```

```{r, fig.width = 35, fig.height = 10}
top20 <- names(sort(taxa_sums(ps.2), decreasing=TRUE))[1:20]
ps.top20 <- transform_sample_counts(ps.2, function(OTU) OTU/sum(OTU))
ps.top20 <- prune_taxa(top20, ps.top20)
plot_bar(ps.top20, x="flea.id", fill="Genus") + facet_grid(~location, scales="free_x",  space = "free_x") +
  ggsave(here("flea_main", "abundance_filtered.pdf"), plot = last_plot(), width = 20, height = 8)+
  labs(x = "Sample ID", y = "Relative Abundance", title = "Relative Abundance after Filtering")
```
```{r, fig.width = 20, fig.height=8}
top20.unfilt <- names(sort(taxa_sums(ps), decreasing=TRUE))[1:20]
ps.top20.unfilt <- transform_sample_counts(ps, function(OTU) OTU/sum(OTU))
ps.top20.unfilt <- prune_taxa(top20.unfilt, ps.top20.unfilt)
plot_bar(ps.top20.unfilt, x="flea.id", fill="Genus") + facet_grid(~location, scales="free_x",  space = "free_x") +
  ggsave(here("flea_main", "abundance_b4_filtered.pdf"), plot = last_plot(), width = 20, height = 8)+
  labs(x = "Sample ID", y = "Relative Abundance", title = "Relative Abundance before Filtering")
```


```{r, fig.width = 20, fig.height = 10}
colorbart <- c("Black", "Red", replicate(18, "Black"))
plot_bar(ps.top20, x="flea.id", fill="Genus") + facet_grid(~location, scales="free_x",  space = "free_x")+
  scale_fill_manual(values = colorbart)+
  labs(x = "Flea ID", title = "Relative Abundance of Bartonella in Flea Sample by Location")+
  ggsave(here("flea_main", "abundance_bartonella_black.pdf"), plot = last_plot(), width = 20, height = 8)
```

```{r, warnings = FALSE}

for(i in 1:18){
  colori <- c(replicate((i-1), "Black"), "Red", replicate((20-i)-1, "Black"))
  name <- paste("abundance_genus_", i, ".pdf")
  plot_bar(ps.top20, x="flea.id", fill="Genus") + facet_grid(~location, scales="free_x",  space = "free_x")+
  scale_fill_manual(values = colori)+
  labs(x = "Flea ID", title = "Relative Abundance in Flea Sample by Location")+
  ggsave(here("flea_main", "abundance_by_genus", name), plot = last_plot(), width = 20, height = 8)
}
```

```{r}

```

Decontam package











